connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java

```diff
@@ -226,20 +235,20 @@ protected final void initializeResources() {
         if (adminListeners == null) {
             log.info("Adding admin resources to main listener");
             adminResourceConfig = resourceConfig;
-            Collection<ConnectResource> adminResources = adminResources();
-            resources.addAll(adminResources);
+            Collection<Class<?>> adminResources = adminResources();
             adminResources.forEach(adminResourceConfig::register);
             configureAdminResources(adminResourceConfig);
         } else if (adminListeners.size() > 0) {
             // TODO: we need to check if these listeners are same as 'listeners'
             // TODO: the following code assumes that they are different
             log.info("Adding admin resources to admin listener");
             adminResourceConfig = new ResourceConfig();
+            adminResourceConfig.register(requestTimeout.binder());
             adminResourceConfig.register(new JacksonJsonProvider());
-            Collection<ConnectResource> adminResources = adminResources();
-            resources.addAll(adminResources);
+            Collection<Class<?>> adminResources = adminResources();
             adminResources.forEach(adminResourceConfig::register);
             adminResourceConfig.register(ConnectExceptionMapper.class);
+            adminResourceConfig.property(ServerProperties.WADL_FEATURE_DISABLE, true);
```

Comment:
This fixes another spurious warning message that gets emitted if the worker is configured with the `admin.listeners` property.

Reply:
this is some light duplication, perhaps the ResourceConfig constructor and the 4 lines accompanying it (request timeout, jackson, exception mapper, and wadl feature disable) can be deduplicated.

Reply:
Good call, done. I've also cleaned up the configuration of the `adminResourceConfig` to hopefully prevent other kinds of duplication-related issues in the future; LMKWYT